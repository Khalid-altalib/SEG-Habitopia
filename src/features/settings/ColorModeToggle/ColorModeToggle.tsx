// React Native
import { Switch } from "react-native";

// Native Base
import { Box, HStack, theme, useColorMode } from "native-base";

// Habitopia
import Text from "@components/Text";
import { TextType } from "types";

/**
 * A switch button for toggling between light and dark mode.
 *
 * @returns The switch button.
 */
const ColorModeToggle = () => {
  const { colorMode, toggleColorMode } = useColorMode();

  return (
    <HStack
      display={"flex"}
      justifyContent={"space-between"}
      alignItems="center"
      paddingTop={3}
    >
      <Box>
        <Text type={TextType.Small}>Light Mode </Text>
        <Text type={TextType.Subtle}>Choose the color mode</Text>
      </Box>
      <Box>
        <Switch
          value={colorMode === "light"}
          onChange={toggleColorMode}
          trackColor={{ true: theme.colors.purple[700] }}
        />
      </Box>
    </HStack>
  );
};

export default ColorModeToggle;
