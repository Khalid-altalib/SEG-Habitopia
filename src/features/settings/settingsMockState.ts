import { SettingsState } from "./settingsSlice";

/**
 * The mock state for the settings redux state
 */

const settingsMockState: SettingsState = {
  settings: {
    email: "test@test.com",
    password: "password",
    notifications: true,
    name: "Test User",
    biography: "Hello, I'm Test User.",
  },
  fetchSettings: {
    loading: false,
    error: "",
  },
  setSettings: {
    loading: false,
    error: "",
  },
};

export default settingsMockState;
