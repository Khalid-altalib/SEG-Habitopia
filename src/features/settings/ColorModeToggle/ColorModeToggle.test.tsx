import { fireEvent, render, waitFor } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import ColorModeToggle from "./ColorModeToggle";
import settingsMockState from "../settingsMockState";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";

describe("ColorModeToggle", () => {
  const mockState = {
    settings: settingsMockState,
  };
  const mockStore = configureStore([thunk])(mockState);

  it("renders correctly", () => {
    const tree = render(
      <TestingWrapperNavigation store={mockStore}>
        <ColorModeToggle />
      </TestingWrapperNavigation>
    );

    expect(tree).toMatchSnapshot();
  });
});
