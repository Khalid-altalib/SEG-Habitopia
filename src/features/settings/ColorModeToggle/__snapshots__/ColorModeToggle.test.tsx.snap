// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorModeToggle renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingTop": 12,
                }
              }
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <Text
                  style={
                    [
                      {
                        "color": "#18181b",
                      },
                      {
                        "fontFamily": undefined,
                        "fontSize": 15,
                      },
                      undefined,
                    ]
                  }
                >
                  Light Mode 
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#18181b",
                      },
                      {
                        "color": "#71717a",
                        "fontFamily": undefined,
                        "fontSize": 15,
                      },
                      undefined,
                    ]
                  }
                >
                  Choose the color mode
                </Text>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                dataSet={{}}
                style={{}}
              >
                <RCTSwitch
                  accessibilityRole="switch"
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  onTintColor="#7e22ce"
                  style={
                    {
                      "height": 31,
                      "width": 51,
                    }
                  }
                  thumbTintColor="white"
                  value={true}
                />
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
